sycamore/scans/file_scan.py

```diff
@@ -133,14 +138,82 @@ def format(self):
         return self._binary_format
 
 
+# Note: We currently handle JSON by reading binary and then parsing it into fields in the _to_document method
+# Ideally we would use the underlying read_json from Ray, but it doesn't support the the include_paths option.
+# Since the path is pretty important for us for lineage, among other things, we take the performance hit of
+# parsing this way, rather than using the pyarrow JSON parser that Ray calls.
 class JsonScan(FileScan):
-    def __init__(self, paths: Union[str, list[str]], *, parallelism: Optional[int] = None, **resource_args):
-        super().__init__(paths, parallelism=parallelism, **resource_args)
+    def __init__(
+        self,
+        paths: Union[str, list[str]],
+        *,
+        properties: Optional[Union[str, list[str]]] = None,
+        parallelism: Optional[int] = None,
+        filesystem: Optional[FileSystem] = None,
+        metadata_provider: Optional[FileMetadataProvider] = None,
+        document_body_field: Optional[str] = None,
+        **resource_args,
+    ):
+        super().__init__(paths, parallelism=parallelism, filesystem=filesystem, **resource_args)
+        self._properties = properties
         self.parallelism = -1 if parallelism is None else parallelism
+        self._metadata_provider = metadata_provider
+        self._document_body_field = document_body_field
 
-    def execute(self) -> "Dataset":
-        json = read_json(paths=self._paths, parallelism=self.parallelism, **self.resource_args)
-        return json
+    def _to_document(self, dict: dict[str, Any]) -> dict[str, Any]:
+        document = Document()
+
+        document.doc_id = str(uuid.uuid1())
+        document.type = "json"
+
+        json_str = dict["bytes"].decode("utf-8")
+        json_dict = json.loads(json_str)
+
+        if self._document_body_field is not None:
+            body = json_dict.pop(self._document_body_field, None)
```

Comment:
here default should be none or json_str?

Reply:
This is the case where the caller specifies a field to use and then it's not present. I think `None` is probably reasonable in that case, though open to suggestions if you feel differently.